spec.go

```diff
@@ -1,13 +1,54 @@
 package torrent
 
 import (
+	"github.com/anacrolix/missinggo"
+	"github.com/anacrolix/torrent/bencode"
 	"github.com/anacrolix/torrent/metainfo"
 	"github.com/anacrolix/torrent/storage"
+	"github.com/pkg/errors"
 )
 
-// Specifies a new torrent for adding to a client. There are helpers for
-// magnet URIs and torrent metainfo files.
-type TorrentSpec struct {
+// Option for the torrent.
+type Option func(*Torrent)
+
+// OptionTrackers set the trackers for the torrent.
+func OptionTrackers(trackers [][]string) Option {
+	return func(t *Torrent) {
+		t.Trackers = trackers
+	}
+}
+
+// OptionDisplayName set the display name for the torrent.
+func OptionDisplayName(dn string) Option {
+	return func(t *Torrent) {
+		t.DisplayName = dn
+	}
+}
+
+// OptionInfo set the info bytes for the torrent.
+func OptionInfo(i []byte) Option {
+	return func(t *Torrent) {
+		t.InfoBytes = i
+	}
+}
+
+// OptionChunk sets the size of the chunks to use for outbound requests
+func OptionChunk(s int) Option {
+	return func(t *Torrent) {
+		t.ChunkSize = s
+	}
+}
+
+// OptionStorage set the storage implementation for the torrent.
+func OptionStorage(s storage.ClientImpl) Option {
```

Comment:
could break the package dependency here by defining its own private interface for storage to satisfy.